velero-plugins/configmap/backup.go

```diff
@@ -0,0 +1,78 @@
+package configmap
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/konveyor/openshift-velero-plugin/velero-plugins/common"
+	"github.com/konveyor/openshift-velero-plugin/velero-plugins/util/openshift"
+	"github.com/sirupsen/logrus"
+	v1 "github.com/vmware-tanzu/velero/pkg/apis/velero/v1"
+	"github.com/vmware-tanzu/velero/pkg/plugin/velero"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// BackupPlugin is a backup item action plugin for Velero.
+type BackupPlugin struct {
+	Log logrus.FieldLogger
+}
+
+// AppliesTo returns a velero.ResourceSelector that applies to configmaps.
+func (p *BackupPlugin) AppliesTo() (velero.ResourceSelector, error) {
+	return velero.ResourceSelector{
+		IncludedResources: []string{"configmaps"},
+	}, nil
+}
+
+// Execute check if configMap should be skipped restore due to
+// belonging to a build pod, which will get regenerated on build restore
+// to avoid restoring cacert from old cluster into new environment
+func (p *BackupPlugin) Execute(item runtime.Unstructured, backup *v1.Backup) (runtime.Unstructured, []velero.ResourceIdentifier, error) {
+
+	p.Log.Info("[cm-backup] Entering ConfigMap backup plugin")
+	configMap := corev1.ConfigMap{}
+	itemMarshal, _ := json.Marshal(item)
+	json.Unmarshal(itemMarshal, &configMap)
+	p.Log.Info(fmt.Sprintf("[cm-backup] ConfigMap: %#v", configMap))
+	// return if not build configmap name
+	if !nameHasBuildSuffix(configMap.Name) {
+		p.Log.Info("[cm-backup] Leaving ConfigMap backup plugin, not a build configmap name")
+		return item, nil, nil
+	}
+	// return if no ownerRef
+	if len(configMap.OwnerReferences) == 0 {
+		p.Log.Info("[cm-backup] Leaving ConfigMap backup plugin, not a build configmap with ownerRef")
+		return item, nil, nil
+	}
+	// return if ownerRef is not pod
+	foundBuildPodRef := false
+	for _, ref := range configMap.OwnerReferences {
+		if ref.Kind == "Pod" {
+			if isBuildPod, err := openshift.IsBuildPod(ref.Name, configMap.Namespace); err != nil {
```

Comment:
here I think is hard to read, would refactor

```go
isBuildPod, err := openshift.IsBuildPod(ref.Name, configMap.Namespace)
if err != nil {
    p.Log.Warnf("[cm-backup] could not determine if ownerRef is build pod: %v", err)
    continue
} if isBuildPod {
    foundBuildPodRef = true
    break
}
```
what you think?

Reply:
Putting the nil check inline vs. separate call and next-line nil check is more of a personal preference. Both styles are common in our code -- I don't have a strong opinion either way. If you do modify this to separate the if from the IsBuildPod call, I would omit that `continue` and just use the `else if isBuildPod` from the current PR -- I think unnecessary continue statements make the code less readable.